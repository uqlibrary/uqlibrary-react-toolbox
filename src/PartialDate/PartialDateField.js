import React from 'react';
import PartialDateForm from './components/PartialDateForm';

export default function PartialDateField(fieldProps) {
    return(<PartialDateForm onChange={ fieldProps.input.onChange } {...fieldProps} />);
}
