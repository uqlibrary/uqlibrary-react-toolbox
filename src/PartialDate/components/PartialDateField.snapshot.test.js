jest.dontMock('./PartialDateField');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import PartialDateField from './PartialDateField';

function setup(props) {
    return shallow(<PartialDateField {...props} />);
}

describe('Redux-form Field PartialDateField snapshots tests', () => {
    it('renders PartialDateField component', () => {
        const props = {
            name: 'partialDate',
            allowPartial: true,
            input: {
                onChange: () => {}
            }
        };

        const wrapper = setup(props);

        let tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});
