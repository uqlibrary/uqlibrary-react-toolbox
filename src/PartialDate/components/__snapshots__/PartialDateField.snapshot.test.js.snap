// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux-form Field PartialDateField snapshots tests renders PartialDateField component 1`] = `
<PartialDateForm
  allowPartial={true}
  className=""
  dateFormat="YYYY-MM-DD"
  floatingTitle="Enter a date"
  floatingTitleRequired={false}
  input={
    Object {
      "onChange": [Function],
    }
  }
  locale={
    Object {
      "dayLabel": "Day",
      "maxNumberCharCode": 57,
      "minNumberCharCode": 48,
      "monthLabel": "Month",
      "validationMessage": Object {
        "day": "Invalid day",
        "month": "Invalid month",
        "year": "Invalid year",
      },
      "yearLabel": "Year",
    }
  }
  months={
    Array [
      "January",
      "February",
      "March",
      "April",
      "May",
      "June",
      "July",
      "August",
      "September",
      "October",
      "November",
      "December",
    ]
  }
  name="partialDate"
  onChange={[Function]}
/>
`;

exports[`Redux-form Field PartialDateField snapshots tests renders PartialDateField component with requiredField class on all fields 1`] = `
<PartialDateForm
  allowPartial={false}
  className="requiredField"
  dateFormat="YYYY-MM-DD"
  floatingTitle="Enter a date"
  floatingTitleRequired={false}
  input={
    Object {
      "onChange": [Function],
    }
  }
  locale={
    Object {
      "dayLabel": "Day",
      "maxNumberCharCode": 57,
      "minNumberCharCode": 48,
      "monthLabel": "Month",
      "validationMessage": Object {
        "day": "Invalid day",
        "month": "Invalid month",
        "year": "Invalid year",
      },
      "yearLabel": "Year",
    }
  }
  months={
    Array [
      "January",
      "February",
      "March",
      "April",
      "May",
      "June",
      "July",
      "August",
      "September",
      "October",
      "November",
      "December",
    ]
  }
  name="partialDate"
  onChange={[Function]}
/>
`;

exports[`Redux-form Field PartialDateField snapshots tests renders PartialDateField component with requiredField class on year field 1`] = `
<PartialDateForm
  allowPartial={true}
  className="requiredField"
  dateFormat="YYYY-MM-DD"
  floatingTitle="Enter a date"
  floatingTitleRequired={false}
  input={
    Object {
      "onChange": [Function],
    }
  }
  locale={
    Object {
      "dayLabel": "Day",
      "maxNumberCharCode": 57,
      "minNumberCharCode": 48,
      "monthLabel": "Month",
      "validationMessage": Object {
        "day": "Invalid day",
        "month": "Invalid month",
        "year": "Invalid year",
      },
      "yearLabel": "Year",
    }
  }
  months={
    Array [
      "January",
      "February",
      "March",
      "April",
      "May",
      "June",
      "July",
      "August",
      "September",
      "October",
      "November",
      "December",
    ]
  }
  name="partialDate"
  onChange={[Function]}
/>
`;
